src/dcmp/dcmp.c

```diff
@@ -603,30 +597,9 @@ static int dcmp_compare_data(
                 /* memory contents are different */
                 rc = 1;
                 copy_src_to_dst = 1;
-                
-                /* if memory contents are different and sync option is on, we want to 
-                 * copy the src bytes to the destination file */ 
-                if (!mfu_copy_opts->do_sync) {
```

Comment:
this one too?  just going on the if condition, haven't studying the code in detail